apps/ogr2ogr_lib.cpp

```diff
@@ -7526,7 +7526,8 @@ static std::unique_ptr<GDALArgumentParser> GDALVectorTranslateOptionsGetParser(
             [psOptions](const std::string &s)
             {
                 psOptions->eGeomOp = GEOMOP_SEGMENTIZE;
-                psOptions->dfGeomOpParam = CPLAtofM(s.c_str());
```

Comment:
It looks like this particular case could have been fixed (more idiomatically?) by adding `.scan<'g', double>()` to the call chain.

What's odd is that (as far as I can see), `.action` still requires you to repeat the string-to-number conversion yourself.

Reply:
> What's odd is that (as far as I can see), `.action` still requires you to repeat the string-to-number conversion yourself.

you can use ``.store_into(reference_to_double_variable)`` as well

Reply:
I missed that. So the original code can simply be replaced by the following and will error if `-simplify` cannot be parsed to a double.

```
    argParser->add_argument("-simplify")
        .metavar("<tolerance>")
        .store_into(psOptions->dfGeomOpParam)
        .action(
            [psOptions](const std::string &)
            {
                psOptions->eGeomOp = GEOMOP_SIMPLIFY_PRESERVE_TOPOLOGY;
            })
        .help(_("Distance tolerance for simplification."));
```